guides/v15-mdc-migration.md

```diff
@@ -0,0 +1,571 @@
+  
+  * If you need your icon to appear at the end of the item (not officially supported by the spec)
+    you can wrap both the text and your icon in a span, e.g.
+
+    ```html
+    <span>
+      <span>Text</span>
+      <mat-icon>end_icon</mat-icon>
+    </span>
+    ```
+    
+* The text in menu items wrap instead of being hidden with an ellipses.
+
+### Option / Optgroup
+
+* Long options now wrap instead of truncating with an ellipsis.
+
+### Paginator
+
+* The form-field inside of `mat-paginator` only supports the `appearance` options offered by the new
+  form-field (`fill` and `outline`).
+
+### Progress Bar
+
+* Hiding the bar with `visibility: hidden` will not hide all internal elements because they apply
+  a `visibility: visible` style. Instead, style it with `opacity: 0` or `display: none`, or remove
+  it completely with `ngIf`.
+
+* Height is always set to 4px and does not get shorter or taller using `height` styles.
+
+### Progress Spinner
+
+* Host element is no longer `display: block` which may affect layout. To fix layout issues, simply
+  add `display: block` back to the element.
+
+### Radio
+
+* Radio-button labels are no longer `width: 100%`. This helps prevent users from accidentally
+  selecting a radio when clicking on whitespace that appears inert on the page.
+
+### Select
+
+* MatSelect no longer aligns the selected option in the dropdown menu with the trigger text.
+
+* Long options now wrap by default instead of truncating.
+
+* Previously the dropdown menu width could be wider than the parent form-field, but now the dropdown
+  is the same width as the form-field
+
+### Slide Toggle
+
+* Accessibility: MDC-based version uses `<button role="switch">` to represent the toggle rather than
+  `<input type="checkbox">`
+
+* Accessibility: The touch target is much larger than the slide toggle; to match the legacy size,
+  you can provide a `-5` density to the slide toggle’s theme mixin:
+
+  ```scss
+  @use '@angular/material' as mat;
+
+  @include mat.slide-toggle-theme(
+    map-merge($theme, (density: -5))
+  );
+  ```
+
+  Without the density mixin, the touch targets collide (see blue outlined targets below) collide:
+
+  TODO(mmalerba): replace this with a working image or embedded example:
+  ![alt_text](../images/slide-toggle-diff.png "Diagram showing overlapping slide-toggle touch targets")
+
+* The label is closer to the enabled toggle
+
+### Slider
+
+* Accessibility: sliders now work with mobile device screen readers.
+
+* To accommodate range sliders, the implementation has changed from the `<mat-slider>` element being
+  the form control to the `<mat-slider>` element containing 1-2 `<input>` elements (the slider
+  "thumbs") that act as the form control(s). The value, associated events (`input`, `change`), and
+  labels (`aria-label`) now live on the `<input>` elements instead.
+
+* Vertical sliders and inverted sliders are no longer supported, as they are not part of the
+  Material Design spec.
```

Comment:
```suggestion
* Vertical sliders and inverted sliders are no longer supported, as they are no longer part of the
  Material Design spec.
```
I don't know if this is worth mentioning, but I strongly suspect we'll be adding vertical sliders back

Reply:
If we're pretty confident about re-adding vertical sliders we could split the bullet to mention that inverted ones are gone and vertical ones are not yet implemented, but planned for a future release

Reply:
Added a TODO